tests/storage/mpathconf_test.py

```diff
@@ -172,3 +201,46 @@ def test_mpath_empty_conf(fake_mpath_conf):
         revision=mpathconf.REVISION_MISSING,
         private=False,
     )
+
+
+@pytest.fixture
+def fake_mpathd_config(request, monkeypatch):
+    def _factory(mpath_output):
+        def mockreturn(*args):
+            return mpath_output
+
+        monkeypatch.setattr(
+            mpathconf.multipath, 'show_config_local', mockreturn)
+
+    return _factory
+
+
+@pytest.mark.parametrize("test_input,section", [
+    (["yes", "no", "no"], "defaults"),
+    (['"yes"', '"no"', '"no"'], "defaults"),
+    (["no", "yes", "no"], "overrides"),
+    (['"no"', '"yes"', '"no"'], "overrides"),
+    (["no", "no", "yes"], "multipath"),
+    (['"no"', '"no"', '"yes"'], "multipath"),
```

Comment:
This is very hard to understand. what is the first "yes", "no", "yes"?

Don't try to create the shortest possible code. Try to create the easier to 
understand code. Test must the simple as possible to ensure they have no bugs.

Reply:
Fair point. Although I don't think it is that difficult to understand, basically the first line of the function is telling you what each value means. And I've seen more complex uses of the parameters in tests. But ok, I'll separate the parser tests into simpler examples, and the checker itself into a separate set of tests and see how it looks.

Reply:
So finally I kept a test with parameters, but simplified the way the test is parametrized so that it is more readable.

And complemented the test with specific tests for the position of the user_friendly_names attribute in a section.

IMO this is nicer than what I did in the previous version of my parametrized code, and nicer than copy&pasting code multiple times for different section. Let me know what you think.